File: bank-app/bank/src/main/java/com/example/bank/controller/AccountController.java
package com.example.bank.controller;

import com.example.bank.dto.AccountCreateRequestDto;
import com.example.bank.dto.AccountResponseDto;
import com.example.bank.entity.Account;
import com.example.bank.service.account.AccountEventService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/accounts")
@RequiredArgsConstructor
public class AccountController {

    private final AccountEventService accountEventService;

    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    public String createAccount(@RequestBody AccountCreateRequestDto accountCreateRequestDto){
        Account account = accountEventService.createAccount(accountCreateRequestDto.getUserId());
        return account.getId().toString();
    }


    @GetMapping("/{accountId}")
    @ResponseStatus(HttpStatus.OK)
    public AccountResponseDto getAccount(@PathVariable String accountId){
        Account account = accountEventService.getAccount(accountId);
        return new AccountResponseDto(account);
    }
}

File: bank-app/otherbank/src/main/kotlin/com/example/otherbank/controller/AccountController.kt
package com.example.otherbank.controller

import com.example.otherbank.dto.AccountCreateRequestDto
import com.example.otherbank.service.AccountService
import lombok.extern.slf4j.Slf4j
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@Slf4j
@RestController
@RequestMapping("/accounts")
class AccountController(val accountService: AccountService) {

    @PostMapping
    fun createAccount(@RequestBody createRequestDto: AccountCreateRequestDto){
        val createAccount = accountService.createAccount(
            createRequestDto.accountNumber,
            createRequestDto.accountName,
            createRequestDto.userName
        )
        println(createAccount.accountName);
        println(createAccount.accountNumber)
    }

}